Makefile

```diff
@@ -100,3 +92,99 @@ tox:
 find: 
 	@read -p "Enter Term: " term; \
 	grep -rnw ./ -e "$$term"
+
+
+### Virtual Environment
+
+.PHONY: install-uv
+install-uv:
+	curl -LsSf https://astral.sh/uv/0.2.25/install.sh | sh
+
+
+.PHONY: reset-venv
+reset-venv:
+	deactivate || true
+	rm -rf .venv
+	uv venv .venv
+
+
+### Dependencies
+
+# When running these commands locally, it is recommended to first reset the environment
+# with: `make reset-venv && source .venv/bin/activate`
+# Otherwise old dependencies might linger around.
+
+# Set EDITABLE to -e to install the package in editable mode outside of CI. This is
+# useful for local development.
+ifdef CI
+EDITABLE=
+else
+EDITABLE=-e
+endif
+
+# Date until which dependencies installed with --exclude-newer must have been released.
+# Dependencies released after this date are ignored.
+EXCLUDE_NEWER_DATE="2024-08-07"
+
+# Install package for local development.
+.PHONY: install-dev
+install-dev:
+	uv pip install ${EDITABLE} . --all-extras --requirement pyproject.toml
+
+
+# Install package with API dependencies only.
+# Should be same command as in the docs with some extra flags for CI:
+# https://docs.lightly.ai/docs/install-lightly#install-the-lightly-python-client
+.PHONY: install-api-only
+install-api-only:
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} --requirement requirements/base.txt
+	uv pip install --exclude-newer ${EXCLUDE_NEWER_DATE} . --no-deps
+
+# Install package with minimal dependencies.
+# 
+# This command is split into multiple steps:
+# 1. Install the dev dependencies to be able to run tests. We don't want to use
+#    the minimal versions for these dependencies.
+# 2. Then we reinstall the package with minimal dependencies.
+# 3. Finally we install setuptools<50. This is necessary for compatibility with old
+#    PyTorch Lightning versions that do not include the correct setuptools dependencies.
+#
+# Explanation of flags:
+# --exclude-newer: We don't want to install dependencies released after that date to
```

Comment:
But then we'll miss breaking dependency updates.

Reply:
Breaking dependencies are tested in `weekly_dependency_test.yml`. I updated it to use `make install-latest`